query/math.go

```diff
@@ -30,6 +30,16 @@ type mathTree struct {
 	Child []*mathTree
 }
 
+var (
+	ErrorIntOverflow     = errors.New("Integer overflow")
+	ErrorFloatOverflow   = errors.New("Float overflow")
```

Comment:
Don't return errors for floats - stick to IEEE 754 standards. If there's an invalid operation, return `NaN`, etc. When encoding this as JSON, encode these special values as strings:
- `"NaN"`
- `"+Inf"`
- `"-Inf"`

Reply:
As we can't handle NaN and Infinity in JSON, falling back to errors.